### PR TITLE
#4 | Add orders table

### PR DESCRIPTION
**Pull request by xonism**

### CODE REVIEW

File: internship-shop/src/main/java/com/twoday/internshipshop/services/WarehouseService.java [hunk 1]
```diff
@@ -24,21 +27,24 @@ public class WarehouseService {
     private String warehouseUrl;
 
     public List<ProductDTO> getAllProducts() {
-        ProductDTO[] productDTOS = getRestTemplate().getForObject(warehouseUrl, ProductDTO[].class);
+        String url = warehouseUrl + "/products";
+        ProductDTO[] productDTOS = getRestTemplate().getForObject(url, ProductDTO[].class);
```
Comment (ArnasPetraitis) [inline internship-shop/src/main/java/com/twoday/internshipshop/services/WarehouseService.java]: I missed last time, but restTemplate should be constructed once in this service. also you can try to set warehouseUrl as base url on restTemplate

File: internship-shop/src/main/java/com/twoday/internshipshop/utils/JsonUtils.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.twoday.internshipshop.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class JsonUtils {
```
Comment (ArnasPetraitis) [inline internship-shop/src/main/java/com/twoday/internshipshop/utils/JsonUtils.java]: 1. is it used?
2. is it duplicated?

File: internship-warehouse/src/main/java/com/twoday/internshipwarehouse/models/Order.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+@Entity
+@Table(name = "orders")
+public class Order {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @NotNull
+    @ManyToOne(cascade = CascadeType.MERGE)
```
Comment (ArnasPetraitis) [inline internship-warehouse/src/main/java/com/twoday/internshipwarehouse/models/Order.java:32]: `CascadeType.MERGE` what does it do?

**Review by ArnasPetraitis — CHANGES_REQUESTED** [review]

File: internship-warehouse/src/main/java/com/twoday/internshipwarehouse/models/Order.java (hunk 3 above)
Comment (xonism) [reply internship-warehouse/src/main/java/com/twoday/internshipwarehouse/models/Order.java:32]: To my understanding, cascades the update operation - for example, if user/product IDs change, then they will change in the order table as well

File: internship-shop/src/main/java/com/twoday/internshipshop/utils/JsonUtils.java (hunk 2 above)
Comment (xonism) [reply internship-shop/src/main/java/com/twoday/internshipshop/utils/JsonUtils.java]: This should not have been in this PR - removed it

File: internship-shop/src/main/java/com/twoday/internshipshop/services/WarehouseService.java (hunk 1 above)
Comment (xonism) [reply internship-shop/src/main/java/com/twoday/internshipshop/services/WarehouseService.java]: Fixed

**Review by ArnasPetraitis — APPROVED** [review]

**Pull request merged**